### ISSUE TITLE
[form] action tag breaks in PR #16029

### ISSUE THREAD

**Issue by jaytaph** [symfony/symfony#16038]
PR #16029 removes the 'action' attribute from a form when the data is empty. This however, is incorrect behaviour according to the HTML specifications, even though URI's according to [RFC 2396](http://www.ietf.org/rfc/rfc2396.txt) section 4.2 states that empty URI's are considered same-document references.

In HTML4, the action attribute is **required**, yet it might contain an empty URI (and thus references to same-document). This is specified in the HTML4 specification (http://www.w3.org/TR/html4/interact/forms.html#h-17.3), section 17.3.

In HTML5 however, the action attribute is **optional**, but if specified, it must point to a ["valid non-empty URL potentially surrounded by spaces"](http://www.w3.org/html/wg/drafts/html/master/semantics.html#attr-fs-action).

This was already discussed in issue #13852.

To recap:

attribute | html4 | html5
------------ | ------------- | -------------
\<form action="http://foo"> | yes | yes
\<form action="">  | yes | no
\<form> | no | yes

This is why originally i've added an `omit_action` option (PR #15995) to the form type: it's not possible to distinguish between a html4 or html5 rendered form, and thus the user must tell explicitly if either A) the action must be added (but should be empty) or B) the action must not be added at all.

**Comment by Tobion** [comment]
Does the html4 form without action not submit to the current document? Or what is the actual bug you experience?

**Comment by Tobion** [comment]
I tested it and html4 without an action is supported by all major browsers. The core team agreed to default to valid html5 beginning with symfony 2.8. If you want to validate against html4, just use action='#' or change your form theme. A new option for this is overkill.
